x/nodes/keeper/reward.go

```diff
@@ -45,41 +112,99 @@ func (k Keeper) RewardForRelaysPerChain(ctx sdk.Ctx, chain string, relays sdk.Bi
 	if isDuringFirstNonCustodialIssue {
 		_, found := k.GetValidator(ctx, address)
 		if !found {
-			ctx.Logger().Error(fmt.Errorf("no validator found for address %s; at height %d\n", address.String(), ctx.BlockHeight()).Error())
+			ctx.Logger().Error(
+				"no validator found",
+				"address", address,
+				"height", ctx.BlockHeight(),
+			)
 			return sdk.ZeroInt()
 		}
 	}
 
-	multiplier := k.GetChainSpecificMultiplier(ctx, chain)
-
-	var coins sdk.BigInt
+	toNode, toFeeCollector :=
+		k.CalculateRelayReward(ctx, chain, relays, validator.GetTokens())
 
-	//check if PIP22 is enabled, if so scale the rewards
-	if isAfterRSCAL {
-		stake := validator.GetTokens()
-		//floorstake to the lowest bin multiple or take ceiling, whicherver is smaller
-		flooredStake := sdk.MinInt(stake.Sub(stake.Mod(k.ServicerStakeFloorMultiplier(ctx))), k.ServicerStakeWeightCeiling(ctx).Sub(k.ServicerStakeWeightCeiling(ctx).Mod(k.ServicerStakeFloorMultiplier(ctx))))
-		//Convert from tokens to a BIN number
-		bin := flooredStake.Quo(k.ServicerStakeFloorMultiplier(ctx))
-		//calculate the weight value, weight will be a floatng point number so cast to DEC here and then truncate back to big int
-		weight := bin.ToDec().FracPow(k.ServicerStakeFloorMultiplierExponent(ctx), Pip22ExponentDenominator).Quo(k.ServicerStakeWeightMultiplier(ctx))
-		coinsDecimal := multiplier.ToDec().Mul(relays.ToDec()).Mul(weight)
-		//truncate back to int
-		coins = coinsDecimal.TruncateInt()
-	} else {
-		coins = multiplier.Mul(relays)
+	// After the delegator upgrade, we compensate a servicer's operator wallet
+	// for the transaction fee of claim and proof.
+	if k.Cdc.IsAfterDelegatorUpgrade(ctx.BlockHeight()) {
+		rewardCost := k.GetRewardCost(ctx)
+		if toNode.LT(rewardCost) {
+			rewardCost = toNode
```

Comment:
Following up on this.

If rewardCost > actualReward, we reward the node with a bigger amount. 

Is this correct? If so, #PUC

Reply:
The actual amount of relay rewards (= servicer's portion 85%) is `toNode`.  If `rewardCost > toNode`, we run `rewardCost = toNode` and reward the servicer with all of the servicer's portion, meaning no rewards for output address and delegators.